frontend/android/fastlane/Fastfile

```diff
@@ -13,26 +13,176 @@
+                -o ~/#{KEYSTORE_PATH} -d #{CREDENTIALS_KEYSTORE_PATH}")
   end
 
-  desc "Submit a new Beta Build to Crashlytics Beta"
-  lane :beta do
-    gradle(task: "clean assembleRelease")
-    crashlytics
-  
-    # sh "your_script.sh"
-    # You can also use other beta testing services here
+  desc "Validate Play Store Key"
+  lane :validate_play_store_key do
+    validate_play_store_json_key(
+      json_key_data: ENV["GOOGLE_SERVICE_ACCOUNT_JSON"]
+    )
   end
 
-  desc "Deploy a new version to the Google Play"
-  lane :deploy do
-    gradle(task: "clean assembleRelease")
-    upload_to_play_store
+  desc "Build Android App"
+  lane :build do |options|
+    version_code = options[:version_code]
+    version_name = options[:version_name]
+    flavor = options[:flavor]
+    build_config_name = options[:build_config_name]
+
+    if [version_name, version_code].include?(nil)
+      raise "'nil' passed as parameter! Aborting..."
+    end
+
+    gradle_system_properties = {
+      # 2GB Gradle + 1GB dex + 2-2.5GB RN < 6GB of circleci resource class medium+
+      :"org.gradle.jvmargs" => "-Xms512m -Xmx2024m",
+      :"org.gradle.daemon" => false
+    }
+
+    if ENV['TOTAL_CPUS']
+      # Gradle uses the wrong cpu count from the host (e.g. 36)
+      gradle_system_properties["org.gradle.workers.max"] = ENV['TOTAL_CPUS']
+    end
+
+    sh("fvm flutter pub run build_runner build --define df_build_config=name=#{build_config_name}")
+
+    gradle(
+      task: "bundle",
+      flavor: flavor,
+      flags: " -Ptarget=lib/main.dart",
+      build_type: "Release",
+      properties: {
+        :VERSION_CODE => version_code,
+        :VERSION_NAME => version_name,
+        :BUILD_CONFIG_NAME => build_config_name,
+        :KEYSTORE_PATH => KEYSTORE_PATH,
+        :KEYSTORE_KEY_ALIAS => flavor == 'Bayern' ? KEYSTORE_KEY_ALIAS_BAYERN : KEYSTORE_KEY_ALIAS_NUERNBERG,
+        :KEYSTORE_PASSWORD => KEYSTORE_PASSWORD,
+        :KEYSTORE_KEY_PASSWORD => flavor == 'Bayern' ? KEYSTORE_KEY_PASSWORD_BAYERN : KEYSTORE_KEY_PASSWORD_NUERNBERG,
+      }.compact,
+      system_properties: gradle_system_properties,
+      print_command: false
+    )
   end
+
+  desc "Upload Android App to Google Play"
+  lane :upload_to_playstore do |options|
+    ensure_env_vars(
+      env_vars: ["GOOGLE_SERVICE_ACCOUNT_JSON"]
+    )
+
+    version_code = options[:version_code]
+    version_name = options[:version_name]
+    production_delivery = options[:production_delivery]
+    build_config_name = options[:build_config_name]
+
+    if [version_name, version_code, production_delivery, build_config_name].include?(nil)
+      raise "'nil' passed as parameter! Aborting..."
+    end
+
+    application_id = sh("app-toolbelt v0 build-config to-json #{build_config_name} android | jq -rj .applicationId ")
+
+    beta_track = application_id == 'de.nrw.it.giz.ehrensache.bayern.android' ? "open-testing" : "beta"
```

Comment:
can you please a comment here that everyone can remember why this check has to be done